[AMD-ZENDNN] Environment variable to enable AMD-ZENDNN plugin

aadwived opened the pull request:
This pull request introduces the environment variable to enable AMD-ZENDNN Plugin
All code changes are under AMD_ZENDNN flag which is enabled for linux builds only.

Authors:
    Aakar Dwivedi ( aakar.dwivedi@amd.com )
    Chandra Kumar Ramasamy ( chandrakumar.ramasamy@amd.com )
    Savan Anadani ( savan.anadani@amd.com )

Signed-off-by: Aakar Dwivedi <Aakar.Dwivedi@amd.com>

aadwived committed:
[AMD-ZENDNN] Environment variable to enable AMD-ZENDNN plugin

This pull request introduces the environment variable to enable AMD-ZENDNN Plugin
All code changes are under AMD_ZENDNN flag which is enabled for linux builds only.

Authors:
    Aakar Dwivedi ( aakar.dwivedi@amd.com )
    Chandra Kumar Ramasamy ( chandrakumar.ramasamy@amd.com )
    Savan Anadani ( savan.anadani@amd.com )

Signed-off-by: Aakar Dwivedi <Aakar.Dwivedi@amd.com>

cramasam commented:
/cc @penpornk

penpornk commented on tensorflow/core/util/port.cc:
```diff
@@ -116,4 +116,31 @@ bool IsMklEnabled() {
+#ifndef AMD_ZENDNN
+  return false;
+#else
+  static bool ZenDNN_enabled = false;
```
Once we get all the integration PRs in and things work fine, we may consider setting the default value to `true` if the plug-in is present. As of now I think it's good to start with false. :)

aadwived replied:
Yes, that is a good idea. Thank you for the quick approval.

penpornk reviewed (approved):
Thank you very much for the PR!

aadwived committed:
[AMD-ZENDNN] Move declaration inside #else condition

- Move absl::once_flag declaration inside #else condition to avoid unused variable issue.

Signed-off-by: Aakar Dwivedi <Aakar.Dwivedi@amd.com>

aadwived commented on tensorflow/core/util/port.cc:
```diff
@@ -116,4 +116,31 @@ bool IsMklEnabled() {
 }
+
+bool IsZenDnnEnabled() {
+  static absl::once_flag once;
```
Resolved. Thank you for the suggestion.

penpornk reviewed (approved):
Thank you for the quick change!

aadwived committed:
Merge branch 'master' into zendnn_tensorflow

aadwived committed:
Add missing new line at the end of the file.
